localstack/services/s3/provider.py

```diff
@@ -1587,6 +1599,79 @@ def select_object_content(
         # avoid a fallback to moto and return the 501 to the client directly instead.
         raise NotImplementedAvoidFallbackError
 
+    def put_bucket_inventory_configuration(
+        self,
+        context: RequestContext,
+        bucket: BucketName,
+        id: InventoryId,
+        inventory_configuration: InventoryConfiguration,
+        expected_bucket_owner: AccountId = None,
+    ) -> None:
+        moto_backend = get_moto_s3_backend(context)
+        get_bucket_from_moto(moto_backend, bucket)
+
+        validate_inventory_configuration(
+            config_id=id, inventory_configuration=inventory_configuration
+        )
+
+        store = self.get_store(context)
+        inventory_configurations = store.bucket_inventory_configurations.setdefault(bucket, {})
+        inventory_configurations[id] = inventory_configuration
+
+    def get_bucket_inventory_configuration(
+        self,
+        context: RequestContext,
+        bucket: BucketName,
+        id: InventoryId,
+        expected_bucket_owner: AccountId = None,
+    ) -> GetBucketInventoryConfigurationOutput:
+        moto_backend = get_moto_s3_backend(context)
+        get_bucket_from_moto(moto_backend, bucket)
+
+        store = self.get_store(context)
+        inventory_configuration = store.bucket_inventory_configurations.get(bucket, {}).get(id)
+        if not inventory_configuration:
+            raise NoSuchConfiguration("The specified configuration does not exist.")
+        return GetBucketInventoryConfigurationOutput(InventoryConfiguration=inventory_configuration)
+
+    def list_bucket_inventory_configurations(
+        self,
+        context: RequestContext,
+        bucket: BucketName,
+        continuation_token: Token = None,
+        expected_bucket_owner: AccountId = None,
+    ) -> ListBucketInventoryConfigurationsOutput:
+        moto_backend = get_moto_s3_backend(context)
+        get_bucket_from_moto(moto_backend, bucket)
+
+        store = self.get_store(context)
+        bucket_inventory_configurations = store.bucket_inventory_configurations.get(bucket, {})
+
+        # bucket_inventory_configurations = sorted(
+        #     bucket_inventory_configurations.values(), key=lambda x: x["Id"]
+        # )
+        # TODO: verify the sorting?
```

Comment:
nit: `list_bucket_xxxx_configurations` are usually sorted by `Id` that is provided on configuration creation. So what you have commented here should be correct.
Similar sorting is being used in `list_bucket_analytics_configurations`. For this maybe we can have a snapshot test to verify.

Reply:
Good catch! I'll add a snapshot test for this, forgot to do it... 😅